### PR TITLE
MMT-1756 - UMM-V version 1.3 -> 1.4

### PR DESCRIPTION
**Pull request by cgokey**
Upgrade from umm-var 1.3 to 1.4

### CODE REVIEW

File: lib/assets/schemas/variables/umm-var-form.json [hunk 1]
```diff
@@ -357,17 +357,32 @@
...
-                  "type": "section",
-                  "htmlClass": "col-6",
-                  "items": [{"key": "/AvgCompressionRateASCII", "field": true, "label": "Avg Compression Rate ASCII"}]
-                },
-                {
-                  "type": "section",
-                  "htmlClass": "col-6",
-                  "items": [{"key": "/AvgCompressionRateNetCDF4", "field": true, "label": "Avg Compression Rate NetCDF4"}]
+                  "type": "UmmMultiItems",
+                  "label": "Average Compression Informations",
```
Comment (cgokey) [inline lib/assets/schemas/variables/umm-var-form.json]: This should not be plural

File: lib/assets/schemas/variables/umm-var-json-schema.json [hunk 2]
```diff
@@ -29,15 +29,15 @@
           "items": {
             "type": "number"
           },
-          "minItems": 0
+          "minItems": 2
```
Comment (cgokey) [inline lib/assets/schemas/variables/umm-var-json-schema.json]: Need to confirm this Eric it is meant to be there and check related tests.

File: lib/assets/schemas/variables/umm-var-json-schema.json [hunk 3]
```diff
@@ -29,15 +29,15 @@
...
         },
         "LonRange": {
           "description": "The LonRange consists of an index range for longitude.",
           "type": "array",
           "items": {
             "type": "number"
           },
-          "minItems": 0
+          "minItems": 2
```
Comment (cgokey) [inline lib/assets/schemas/variables/umm-var-json-schema.json]: Likewise check Erich on this one.

File: spec/features/variable_drafts/forms/validation_spec.rb [hunk 4]
```diff
@@ -343,4 +339,25 @@
...
+        fill_in 'variable_draft_draft_characteristics_index_ranges_lon_range_0', with: '1'
+        within '.nav-top' do
+          click_on 'Save'
+        end
+      end
+
+      it 'displays validation error messages' do
+        expect(page).to have_css('#variable_draft_draft_characteristics_index_ranges_lat_range_error', text: 'Lat Range has too few items')
+        expect(page).to have_css('#variable_draft_draft_characteristics_index_ranges_lon_range_error', text: 'Lon Range has too few items')
+      end
```
Comment (charleshuang80) [inline spec/features/variable_drafts/forms/validation_spec.rb]: I feel like we should add another item to one or both ranges, to test that the error goes away.

File: app/assets/javascripts/validation.coffee [hunk 5]
```diff
@@ -349,6 +347,13 @@ $(document).ready ->
               params: {}
             errors.push newError
 
+  validateLatLonRanges = (errors) ->
+    if $("[id^=variable_draft_draft_characteristics_index_ranges_lat_range_]").length > 0
```
Comment (charleshuang80) [inline app/assets/javascripts/validation.coffee]: It looks like all you are doing is visiting the field if the parent id exists on the page. So those fields will be visited every time the page is validated, right? We usually only validate a field if it has already been visited, but here it seems the field will be marked as visited even if the user hasn't visited it.

File: spec/features/variable_drafts/forms/validation_spec.rb (hunk 4 above)
Comment (htranho) [reply spec/features/variable_drafts/forms/validation_spec.rb]: Yes, will add that test

File: app/assets/javascripts/validation.coffee (hunk 5 above)
Comment (htranho) [reply app/assets/javascripts/validation.coffee]: Added fix in method validateLatLonRanges

File: app/assets/javascripts/validation.coffee [hunk 6]
```diff
@@ -349,6 +347,15 @@ $(document).ready ->
               params: {}
             errors.push newError
 
+  validateLatLonRanges = ->
```
Comment (charleshuang80) [inline app/assets/javascripts/validation.coffee]: do you think this can be done in the `visitField` function on line 536?

Comment (htranho) [reply app/assets/javascripts/validation.coffee]: Now moved to visitField

**Pull request merged**